src/main/java/net/sf/jabref/logic/importer/fetcher/MedlineFetcher.java

```
@@ -0,0 +1,202 @@
+    private int retstart;
+    private String ids = "";
+
+    private void addID(String id) {
+        if (ids.isEmpty()) {
+            ids = id;
+        } else {
+            ids += "," + id;
+        }
+    }
+
+    /**
+     * Removes all comma's in a given String
+     *
+     * @param query input to remove comma's
+     * @return input without comma's
+     */
+    private static String replaceCommaWithAND(String query) {
+        return query.replaceAll(", ", " AND ").replaceAll(",", " AND ");
+    }
+    /**
+     * Gets the initial list of ids
+     */
+    private void getIds(String term, int start, int pacing) {
+        boolean doCount = true;
+        try {
+            URL ncbi = createSearchUrl(term, start, pacing);
+            // get the ids
+            BufferedReader in = new BufferedReader(new InputStreamReader(ncbi.openStream()));
+            String inLine;
+            while ((inLine = in.readLine()) != null) {
+
+                // get the count
+                Matcher idMatcher = ID_PATTERN.matcher(inLine);
+                if (idMatcher.find()) {
+                    addID(idMatcher.group(1));
+                }
+                Matcher retMaxMatcher = RET_MAX_PATTERN.matcher(inLine);
+                if (retMaxMatcher.find()) {
+                    retmax = Integer.parseInt(retMaxMatcher.group(1));
+                }
+                Matcher retStartMatcher = RET_START_PATTERN.matcher(inLine);
+                if (retStartMatcher.find()) {
+                    retstart = Integer.parseInt(retStartMatcher.group(1));
+                }
+                Matcher countMatcher = COUNT_PATTERN.matcher(inLine);
+                if (doCount && countMatcher.find()) {
+                    count = Integer.parseInt(countMatcher.group(1));
+                    doCount = false;
+                }
+            }
+        } catch (URISyntaxException | MalformedURLException e) { // new URL() failed
+            LOGGER.warn("Bad url", e);
```

Rethrow as FetcherException instead of failing silently.